textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/PhraseMatcherTest.java

```diff
@@ -278,5 +279,180 @@ public void testWordInMultipleFieldsQueryWithStopWords2() throws Exception {
+        }
+        schemaAttributes[schemaAttributes.length - 1] = SchemaConstants.SPAN_LIST_ATTRIBUTE;
+
+        IField [] fields1 = {new IntegerField(14347980), new TextField(""), new TextField("CHRONIC MENINGOCOCCEMIA; EPIDEMIOLOGY, DIAGNOSIS AND TREATMENT."),
+    			new TextField("D S BLOOM"), new StringField("103 Aug, 1965"), new TextField("California medicine"), new TextField("DRUG THERAPY, MENINGOCOCCAL INFECTIONS, PENICILLIN G, SULFONAMIDES"),
+    			new TextField("Drug Therapy, Meningococcal Infections, Penicillin G, Sulfonamides"), 
+    			new TextField("This report describes four cases of chronic meningococcemia with the characteristic manifestations of recurrent episodes of "
+    					+ "fever, chills, night sweats, headache and anorexia, associated with skin rash and arthralgias. The diagnosis was established in all instances by blood culture. Administration "
+    					+ "of sulfonamides in three cases and penicillin in the fourth resulted in prompt recovery. The recent finding of a strain of sulfonamide-resistant meningococci, however, indicates "
+    					+ "that antibiotic-sensitivity tests should be carried out in all cases of meningococcal disease. While waiting for the results of such tests to be reported, the clinician should "
+    					+ "initiate treatment with large doses of a sulfonamide and penicillin in combination."), new DoubleField(0.664347980), new ListField<>(list)};
+
+        ITuple tuple1 = new DataTuple(new Schema(schemaAttributes), fields1);
+        List<ITuple> expectedResultList = new ArrayList<>();
+        expectedResultList.add(tuple1);
+
+        //Perform Query
+        IPredicate predicate = new KeywordPredicate(query, attributeList, DataConstants.KeywordOperatorType.PHRASE, MedAnalyzer, MedDataStore);
+        KeywordMatcher = new KeywordMatcher(predicate);
+        KeywordMatcher.open();
+
+        List<ITuple> results = new ArrayList<>();
+        ITuple nextTuple = null;
+
+        while ((nextTuple = KeywordMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+        //Perform Check
+        boolean contains = TestUtils.containsAllResults(expectedResultList, results);
+        Assert.assertTrue(contains);
+    }
+    
+    /**
+     * Verifies: Query with Stop Words match corresponding phrases in the document
+     * Used to cause exception if there is a special symbol with the words, ex: big(
+     * @throws Exception with Medline data
+     */
+    @Test
+    public void testWordInMultipleFieldsQueryWithStopWords4() throws Exception {
+    	DataStore MedDataStore = new DataStore("../index/test", keywordTestConstants.SCHEMA_MEDLINE);
+        Analyzer MedAnalyzer = new StandardAnalyzer();
+        DataWriter MedDataWriter = new DataWriter(MedDataStore, MedAnalyzer);
+        MedDataWriter.clearData();
+    	MedDataWriter.writeData(keywordTestConstants.getSampleMedlineRecord());
+        //Prepare Query
+        String query = "x-ray";
```

Comment:
For the record. This query fails in the old query engine because the inconsistency between tokenization of Lucene's analyzer and keywordMatcher.